org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -347,8 +347,17 @@ private void createBufferFD() throws SSLException {
             throw new SSLException("Unable to enable SSL Handshake Callback on this SSLFDProxy instance.");
         }
 
-        // Pass this ssl_fd to the session object so that we can use
-        // SSL methods to invalidate the session.
+        if (alpn_protocols != null) {
+            byte[] wire_data = getALPNWireData();
+            if (wire_data == null) {
+                throw new RuntimeException("JSSEngine.init(): ALPN wire data is NULL but alpn_protocols is non-NULL.");
+            }
+
+            ret = SSL.SetNextProtoNeg(ssl_fd, wire_data);
+            if (ret != SSL.SECSuccess) {
+                throw new RuntimeException("JSSEngine.init(): Unable to set ALPN protocol list: " + errorText(PR.GetError()) + " " + ret);
```

Comment:
Per other comments, this appears to be calling `SSL_SetNextProtoNego` for both client and server.  And that is the correct behaviour.  But we need to handle the address the NSS protocol ordering quirks.